test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj

```diff
@@ -362,6 +363,38 @@
          (finally
            (jruby-testutils/return-instance jruby-service jruby-instance :settings-plumbed-test)))))))
 
+(deftest jruby-environment-vars-test
+  (testing "Make sure that the environment variables whitelisted in puppetserver.conf are being set"
+    (tk-testutils/with-app-with-config
+      app
+      [jruby/jruby-puppet-pooled-service
+       profiler/puppet-profiler-service
+       jruby-utils/jruby-pool-manager-service]
+      (let [tmp-conf (ks/temp-file "puppetserver" ".conf")]
+        (spit tmp-conf
+              "environment-vars: { \"FOO\": ${HOME} }")
+        (jruby-testutils/jruby-puppet-tk-config
+          (jruby-testutils/jruby-puppet-config
+            (merge
+              {:ruby-load-path   jruby-testutils/ruby-load-path
+               :gem-home         jruby-testutils/gem-home
+               :master-conf-dir  jruby-testutils/conf-dir
+               :master-code-dir  jruby-testutils/code-dir
+               :master-var-dir   jruby-testutils/var-dir
+               :master-run-dir   jruby-testutils/run-dir
+               :master-log-dir   jruby-testutils/log-dir}
+              (tk-config/load-config (.getPath tmp-conf))))))
+      (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+            jruby-instance (jruby-testutils/borrow-instance jruby-service :test)
```

Comment:
Not critical for what this test is doing, but I think it would be good to have the last argument here match the last argument in the corresponding `return-instance` call below - i.e., make them both `:test` or `:settings-plumbed-test`.